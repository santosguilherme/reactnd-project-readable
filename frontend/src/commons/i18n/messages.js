export default {
    'pt-BR': {
        'LABELS': {
            'AT': 'Em',
            'ALL_CATEGORIES': 'Todas',
            'ALL_POSTS': 'Todos',
            'APP_BAR_POST_LIST': '{category} posts',

            'CANCEL': 'Cancelar',
            'CATEGORY': 'Categoria',
            'COMMENT': 'Comentário',
            'COMMENT_AUTHOR': 'Autor',
            'COMMENT_AUTHOR_FORM_PLACEHOLDER': 'Digite o nome do autor do comentário',
            'COMMENT_BODY': 'Comentário',
            'COMMENT_BODY_FORM_PLACEHOLDER': 'Digite o texto do comentário',
            'COMMENT_FORM_SAVE': 'Salvar comentário',
            'COMMENT_VOTE_SCORE_PLURAL': '{count, plural, =0 {Votos} one {Voto} other {Votos}}',
            'CONFIRM_BUTTON_CANCEL': 'Não',
            'CONFIRM_BUTTON_OK': 'Sim, eu confirmo',
            'CONFIRM_TITLE': 'Atenção',

            'POST': 'Post',
            'POST_AUTHOR': 'Autor',
            'POST_AUTHOR_FORM_PLACEHOLDER': 'Digite o nome do autor do post',
            'POST_BODY': 'Texto',
            'POST_BODY_FORM_PLACEHOLDER': 'Digite o texto do post',
            'POST_CATEGORY': 'Categoria',
            'POST_CATEGORY_FORM_PLACEHOLDER': 'Selecione a categoria do post',
            'POST_COMMENTS': '{count, plural, =0 {comentários} one {comentário} other {comentários}}',
            'POST_COMMENTS_COUNT': '{count, plural, =0 {Nenhum comentário} one {# comentário} other {# comentários}}',
            'POST_FORM_SAVE': 'Salvar post',
            'POST_TIMESTAMP': 'Data de criação',
            'POST_TITLE': 'Título',
            'POST_TITLE_FORM_PLACEHOLDER': 'Digite o título do post',
            'POST_VOTE_SCORE': 'Votos',
            'POST_VOTE_SCORE_PLURAL': '{count, plural, =0 {Votos} one {Voto} other {Votos}}',
            'POSTS_NOT_FOUND': 'Não foi encontrado nenhum post, tente refazer a busca ou criar um novo...',

            'MENU_OPTION_EDIT': 'Editar',
            'MENU_OPTION_DELETE': 'Remover',

            'NEW_COMMENT': 'Novo comentário',

            'ORDER_BY': 'Ordenação',
            'OK': 'Ok',

            'VIEW_POST_DETAILS': 'Visualizar post'
        },
        'MESSAGES': {
            'COMMENT_CREATE_SUCCESS': 'Comentário criado com sucesso!',
            'COMMENT_DELETE_SUCCESS': 'Comentário removido com sucesso!',
            'COMMENT_VOTE_SUCCESS': 'Voto realizado com sucesso!',
            'COMMENT_UPDATE_SUCCESS': 'Comentário atualizado com sucesso!',

            'HTTP_NOT_FOUND': 'Não foi possível conectar-se ao servidor.',

            'LOADING': 'Aguarde um instante...',

            'POST_CREATE_SUCCESS': 'Post criado com sucesso!',
            'POST_DELETE_SUCCESS': 'Post removido com sucesso!',
            'POST_VOTE_SUCCESS': 'Voto realizado com sucesso!',
            'POST_UPDATE_SUCCESS': 'Post atualizado com sucesso!',
            'POST_NOT_FOUND': 'O post não foi encontrado ou já foi removido',

            'REQUIRED_FIELD': 'Campo obrigatório',
            'REMOVE_COMMENT_CONFIRM': 'Você confirma que deseja remover o comentário?',
            'REMOVE_POST_CONFIRM': 'Você confirma que deseja remover o post?',

            'UNKNOWN_ERROR': 'Ocorreu um erro não esperado!'
        }
    }
};

/**
 * https://egghead.io/lessons/react-convert-a-hard-coded-string-using-react-intl-formattedmessage
 */
export const flattenMessages = (nestedMessages, prefix = '') => {
    return Object.keys(nestedMessages).reduce((messages, key) => {
        let value = nestedMessages[key];
        let prefixedKey = prefix ? `${prefix}.${key}` : key;

        if (typeof value === 'string') {
            messages[prefixedKey] = value;
        } else {
            Object.assign(messages, flattenMessages(value, prefixedKey));
        }

        return messages;
    }, {});
};