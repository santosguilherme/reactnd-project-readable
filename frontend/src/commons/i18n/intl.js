import {IntlProvider, addLocaleData} from 'react-intl';
import pt from 'react-intl/locale-data/pt';

import {DEFAULT_LOCALE, getLocale} from '../locale/locale';

import messages, {flattenMessages} from './messages';


function configureLocale(renderCallback) {
    addLocaleData(pt);
    renderCallback();
}

export function configureReactIntlPolyfill(renderCallback) {
    if (window.Intl) {
        configureLocale(renderCallback);
        return;
    }

    require.ensure([
            'intl',
            'intl/locale-data/jsonp/pt.js'
        ],
        require => {
            require('intl');
            require('intl/locale-data/jsonp/pt.js');

            configureLocale(renderCallback);
        }
    );
}

export function getIntlProviderConfig() {
    const locale = getLocale();
    const messages = getFlattedLocaleMessages();

    return {locale, messages};
}

export function getFlattedLocaleMessages() {
    let locale = getLocale();
    return flattenMessages(messages[locale] || messages[DEFAULT_LOCALE]);
}

export function getFormattedMessage(id) {
    const config = getIntlProviderConfig();

    return new IntlProvider(config, {})
        .getChildContext()
        .intl
        .formatMessage({id});
}