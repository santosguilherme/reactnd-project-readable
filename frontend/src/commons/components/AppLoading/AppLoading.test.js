import React from 'react';
import {shallow} from 'enzyme';
import configureStore from 'redux-mock-store';

import {mountWithIntl} from '../../i18n/intl-enzyme-test-helper';

import {Provider} from 'react-redux';

import ConnectedAppLoading, {AppLoading} from './AppLoading';


it('renders without crashing', () => {
    const wrapper = shallow(<AppLoading/>);

    expect(wrapper).toMatchSnapshot();
});

it('Should render AppLoading', () => {
    const wrapper = shallow(<AppLoading show={true}/>);

    expect(wrapper).toMatchSnapshot();
});

it('The rendered component must have the classes', () => {
    const test = mountWithIntl(<AppLoading show={true}/>);

    expect(test.find('.app-loading').length).toBeTruthy();
    expect(test.find('.app-loading__message').length).toBeTruthy();
});

it('Test the AppLoading with connect() from react-redux', () => {
    const mockStore = configureStore([]);
    const store = mockStore({
        loading: true
    });

    const wrapper = shallow(
        <Provider store={store}>
            <ConnectedAppLoading/>
        </Provider>
    );

    expect(wrapper).toMatchSnapshot();
});