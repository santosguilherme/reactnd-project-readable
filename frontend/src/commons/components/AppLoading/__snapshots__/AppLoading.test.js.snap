// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render AppLoading 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppLoading
    show={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="app-loading"
      >
        <withStyles(CircularProgress) />
        <withStyles(Typography)
          align="center"
          className="app-loading__message"
          type="body2"
        >
          <FormattedMessage
            id="MESSAGES.LOADING"
            values={Object {}}
          />
        </withStyles(Typography)>
      </div>,
      "test": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <withStyles(CircularProgress) />,
          <withStyles(Typography)
            align="center"
            className="app-loading__message"
            type="body2"
          >
            <FormattedMessage
              id="MESSAGES.LOADING"
              values={Object {}}
            />
          </withStyles(Typography)>,
        ],
        "className": "app-loading",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "align": "center",
            "children": <FormattedMessage
              id="MESSAGES.LOADING"
              values={Object {}}
            />,
            "className": "app-loading__message",
            "type": "body2",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "id": "MESSAGES.LOADING",
              "values": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="app-loading"
        >
          <withStyles(CircularProgress) />
          <withStyles(Typography)
            align="center"
            className="app-loading__message"
            type="body2"
          >
            <FormattedMessage
              id="MESSAGES.LOADING"
              values={Object {}}
            />
          </withStyles(Typography)>
        </div>,
        "test": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <withStyles(CircularProgress) />,
            <withStyles(Typography)
              align="center"
              className="app-loading__message"
              type="body2"
            >
              <FormattedMessage
                id="MESSAGES.LOADING"
                values={Object {}}
              />
            </withStyles(Typography)>,
          ],
          "className": "app-loading",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "align": "center",
              "children": <FormattedMessage
                id="MESSAGES.LOADING"
                values={Object {}}
              />,
              "className": "app-loading__message",
              "type": "body2",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "id": "MESSAGES.LOADING",
                "values": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Test the AppLoading with connect() from react-redux 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Connect(AppLoading) />
  </Provider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppLoading
    show={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="app-loading"
      >
        <withStyles(CircularProgress) />
        <withStyles(Typography)
          align="center"
          className="app-loading__message"
          type="body2"
        >
          <FormattedMessage
            id="MESSAGES.LOADING"
            values={Object {}}
          />
        </withStyles(Typography)>
      </div>,
      "test": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <withStyles(CircularProgress) />,
          <withStyles(Typography)
            align="center"
            className="app-loading__message"
            type="body2"
          >
            <FormattedMessage
              id="MESSAGES.LOADING"
              values={Object {}}
            />
          </withStyles(Typography)>,
        ],
        "className": "app-loading",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "align": "center",
            "children": <FormattedMessage
              id="MESSAGES.LOADING"
              values={Object {}}
            />,
            "className": "app-loading__message",
            "type": "body2",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "id": "MESSAGES.LOADING",
              "values": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="app-loading"
        >
          <withStyles(CircularProgress) />
          <withStyles(Typography)
            align="center"
            className="app-loading__message"
            type="body2"
          >
            <FormattedMessage
              id="MESSAGES.LOADING"
              values={Object {}}
            />
          </withStyles(Typography)>
        </div>,
        "test": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <withStyles(CircularProgress) />,
            <withStyles(Typography)
              align="center"
              className="app-loading__message"
              type="body2"
            >
              <FormattedMessage
                id="MESSAGES.LOADING"
                values={Object {}}
              />
            </withStyles(Typography)>,
          ],
          "className": "app-loading",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "align": "center",
              "children": <FormattedMessage
                id="MESSAGES.LOADING"
                values={Object {}}
              />,
              "className": "app-loading__message",
              "type": "body2",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "id": "MESSAGES.LOADING",
                "values": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
